### PR TITLE
ON-13809 # Added validationEndpoint type

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: flow-typed/forms.js [hunk 1]
```diff
@@ -443,6 +458,7 @@ declare type Form = {
   tags: Array<string>,
   createdAt: string,
   updatedAt: string,
+  validationEndpoint?: FormValidationEndpoint,
```
Comment (mymattcarroll) [inline flow-typed/forms.js:461]:
```suggestion
  serverValidation?: FormServerValidation,
```

File: flow-typed/forms.js [hunk 2]
```diff
@@ -422,6 +422,21 @@ declare type FormPostSubmissionAction =
   | 'CLOSE'
   | 'FORMS_LIBRARY'
 
+type FormValidationEndpoint =
+  | {
+      type: 'CALLBACK',
```
Comment (mymattcarroll) [inline flow-typed/forms.js:427]:
```suggestion
      type: 'URL',
```

File: flow-typed/forms.js [hunk 3]
```diff
@@ -422,6 +422,21 @@ declare type FormPostSubmissionAction =
   | 'CLOSE'
   | 'FORMS_LIBRARY'
 
+type FormValidationEndpoint =
```
Comment (mymattcarroll) [inline flow-typed/forms.js:425]:
```suggestion
type FormServerValidation =
```

File: flow-typed/forms.js (hunk 2 above)
Comment (Zaxist) [reply flow-typed/forms.js:427]: This is to make consistent with submission events

**Pull request merged**